pysph/sph/bc/inlet_outlet_manager.py

```diff
@@ -0,0 +1,318 @@
+"""
+Inlet Outlet Manager
+"""
+# Copyright (c) 2015-2017, Prabhu Ramachandran
+# License: BSD
+
+from pysph.sph.equation import Equation
+
+
+class InletInfo(object):
+    def __init__(self, pa_name, normal, refpoint, freesurface=False,
+                 basenormal=None, equations=None):
+        """Create object with information of inlets, all the others parameters
+           which are not passed here get evaluated by `InletOutletManager`once
+           the inlet is created.
+
+        Parameters
+        ----------
+
+        pa_name : str
+            Name of the inlet
+        normal : list
+            Components of normal (float)
+        refpoint : list
+            Point at the fluid-inlet interface (float)
+        freesurface : bool
+            if True, the inlet will need as basenormal
+        basenormal : list
+            List of normal of the base wall (float)
+        equations : list
+            List of equations (optional)
+        """
+        self.pa_name = pa_name
+        self.normal = normal
+        self.refpoint = refpoint
+        self.length = 0.0
+        self.freesurface = freesurface
+        self.dx = 0.1
+        self.bound = []
+        self.domain = []
+        self.basenormal = [0.0, 0.0, 0.0] if basenormal is None else basenormal
+        self.equations = [] if equations is None else equations
+
+
+class OutletInfo(InletInfo):
+    """Create object with information of outlet
+
+    The name is kept different for distinction only.
+    """
+    pass
+
+
+class InletOutletManager(object):
+    def __init__(self, fluid_arrays, inletinfo, outletinfo, extraeqns=None):
+        """Create the object to manage inlet outlet boundary conditions.
+           Most of the variables are evaluated after the scheme and particles
+           are created.
+
+        Parameters
+        -----------
+
+        fluid_arrays : list
+            List of fluid particles array names (str)
+        inletinfo : list
+            List of inlets (InletInfo)
+        outletinfo : list
+            List of outlet (OutletInfo)
+        extraeqns : dict
+            Dict of custom equations
+        """
+        self.fluids = fluid_arrays
+        self.dim = None
+        self.kernel = None
+        self.inlets = [x.pa_name for x in inletinfo]
+        self.outlets = [x.pa_name for x in outletinfo]
+        self.inlet_pairs = {}
+        self.outlet_pairs = {}
+        self.inletinfo = inletinfo
+        self.outletinfo = outletinfo
+        self.extraeqns = {} if extraeqns is None else extraeqns
+        self.active_stages = []
+
+    def update_dx(self, dx):
+        """Update the discretisation length
+
+        Parameters
+        -----------
+
+        dx : float
+            The discretisation length
+        """
+        all_info = self.inletinfo + self.outletinfo
+        for info in all_info:
+            info.dx = dx
+
+    def update_inlet_outlet_info(self, pa):
```

Comment:
Is this really a public method or an internal method that is only used by this class and not by a generic user?  If so, this could be made a private method.

Reply:
done